File: src-tauri/src/cmd/wework_accounts.rs
//! WeCom saved accounts IPC (SQLite).

use tauri::State;

use crate::context::channel::wework_accounts::{WeworkAccountsStore, WeworkSavedAccountDto};

#[tauri::command]
pub fn wework_list_accounts(
    store: State<'_, WeworkAccountsStore>,
) -> Result<Vec<WeworkSavedAccountDto>, String> {
    crate::log_cmd_result!("cmd.wework.list_accounts", store.list_accounts())
}

#[tauri::command]
pub fn wework_upsert_account(
    store: State<'_, WeworkAccountsStore>,
    account: WeworkSavedAccountDto,
) -> Result<WeworkSavedAccountDto, String> {
    let account_id = account.id.clone();
    crate::log_cmd_result!(
        "cmd.wework.upsert_account",
        store.upsert_account(account),
        "id={account_id}"
    )
}

#[tauri::command]
pub fn wework_delete_account(
    store: State<'_, WeworkAccountsStore>,
    id: String,
) -> Result<(), String> {
    crate::log_cmd_result!(
        "cmd.wework.delete_account",
        store.delete_account(&id),
        "id={id}"
    )
}

#[tauri::command]
pub fn wework_get_active_account_id(
    store: State<'_, WeworkAccountsStore>,
) -> Result<Option<String>, String> {
    crate::log_cmd_result!(
        "cmd.wework.get_active_account_id",
        store.get_active_account_id()
    )
}

#[tauri::command]
pub fn wework_set_active_account_id(
    store: State<'_, WeworkAccountsStore>,
    id: Option<String>,
) -> Result<(), String> {
    let active_id = id.as_deref().unwrap_or("none").to_string();
    crate::log_cmd_result!(
        "cmd.wework.set_active_account_id",
        store.set_active_account_id(id.as_deref()),
        "id={active_id}"
    )
}

#[tauri::command]
pub fn wework_mark_contacts_synced(
    store: State<'_, WeworkAccountsStore>,
    wework_user_id: String,
    synced_at: i64,
) -> Result<(), String> {
    crate::log_cmd_result!(
        "cmd.wework.mark_contacts_synced",
        store.mark_contacts_synced(&wework_user_id, synced_at),
        "wework_user_id={wework_user_id} synced_at={synced_at}"
    )
}

#[tauri::command]
pub fn wework_contacts_synced(
    store: State<'_, WeworkAccountsStore>,
    wework_user_id: String,
) -> Result<bool, String> {
    crate::log_cmd_result!(
        "cmd.wework.contacts_synced",
        store.contacts_synced(&wework_user_id),
        "wework_user_id={wework_user_id}"
    )
}

File: packages/shared/src/tauri-bridge/cmd/wework-accounts.ts
import { invokeWrapper } from "./invoke";
import { TauriCmd } from "../enums/tauri-cmd";

export interface WeworkAccountConfigDto {
  weworkExePath?: string;
  weworkVersion?: string;
  weworkSmart?: boolean;
  weworkInitWaitSeconds?: number;
}

export interface WeworkSavedAccountDto {
  id: string;
  label: string;
  config: WeworkAccountConfigDto;
  createdAt: number;
  lastConnectedAt?: number;
  weworkUserId?: string;
  contactsSynced: boolean;
  contactsSyncedAt?: number;
}

export async function weworkListAccounts(): Promise<WeworkSavedAccountDto[]> {
  return invokeWrapper<WeworkSavedAccountDto[]>(TauriCmd.WeworkListAccounts);
}

export async function weworkUpsertAccount(
  account: WeworkSavedAccountDto
): Promise<WeworkSavedAccountDto> {
  return invokeWrapper<WeworkSavedAccountDto>(TauriCmd.WeworkUpsertAccount, { account });
}

export async function weworkDeleteAccount(id: string): Promise<void> {
  await invokeWrapper<void>(TauriCmd.WeworkDeleteAccount, { id });
}

export async function weworkGetActiveAccountId(): Promise<string | null> {
  return invokeWrapper<string | null>(TauriCmd.WeworkGetActiveAccountId);
}

export async function weworkSetActiveAccountId(id: string | null): Promise<void> {
  await invokeWrapper<void>(TauriCmd.WeworkSetActiveAccountId, { id });
}

export async function weworkMarkContactsSynced(
  weworkUserId: string,
  syncedAt: number
): Promise<void> {
  await invokeWrapper<void>(TauriCmd.WeworkMarkContactsSynced, {
    weworkUserId,
    syncedAt
  });
}

export async function weworkContactsSynced(weworkUserId: string): Promise<boolean> {
  return invokeWrapper<boolean>(TauriCmd.WeworkContactsSynced, { weworkUserId });
}
